src/Spago/Command/Bundle.purs

```diff
@@ -47,38 +53,85 @@ run = do
       BundleBrowser, BundleApp -> "--format=iife"
       _, _ -> "--format=esm"
 
-    -- See https://github.com/evanw/esbuild/issues/1921
-    nodePatch = case opts.platform of
-      BundleNode -> [ "--banner:js=import __module from \'module\';import __path from \'path\';import __url from \'url\';const require = __module.createRequire(import.meta.url);const __dirname = __path.dirname(__url.fileURLToPath(import.meta.url));const __filename=new URL(import.meta.url).pathname" ]
-      _ -> []
+    onlyForNode s = case opts.platform of
+      BundleNode -> s
+      BundleBrowser -> ""
 
-    output = case workspace.buildOptions.output of
-      Nothing -> "output"
-      Just o -> o
+    output = workspace.buildOptions.output # fromMaybe "output"
     -- TODO: we might need to use `Path.relative selected.path output` instead of just output there
     mainPath = withForwardSlashes $ Path.concat [ output, opts.module, "index.js" ]
 
-    shebang = case opts.platform of
-      BundleNode -> "#!/usr/bin/env node\n\n"
-      _ -> ""
-
     { input, entrypoint } = case opts.type of
-      BundleApp -> { entrypoint: [], input: Cmd.StdinWrite (shebang <> "import { main } from './" <> mainPath <> "'; main();") }
-      BundleModule -> { entrypoint: [ mainPath ], input: Cmd.StdinNewPipe }
+      BundleApp ->
+        { entrypoint: []
+        , input: Cmd.StdinWrite $ fold [ onlyForNode "#!/usr/bin/env node\n\n", "import { main } from './", mainPath, "';main();" ]
+        }
+      BundleModule ->
+        { entrypoint: [ mainPath ]
+        , input: Cmd.StdinNewPipe
+        }
+
     execOptions = Cmd.defaultExecOptions { pipeStdin = input }
 
-    args =
-      [ "--bundle"
-      , "--outfile=" <> outfile
-      , "--platform=" <> show opts.platform
-      -- See https://github.com/evanw/esbuild/issues/1051
-      , "--loader:.node=file"
-      , format
-      ] <> opts.extraArgs <> minify <> sourceMap <> entrypoint <> nodePatch
+    banner = fold
+      [ spagoGeneratedFilePrefix
+      , " v"
+      , BuildInfo.packages."spago-bin"
+      , " */"
+      , onlyForNode nodeTargetPolyfill
+      ]
+
+    args = fold
+      [ [ "--bundle"
+        , "--outfile=" <> outfile
+        , "--platform=" <> show opts.platform
+        , "--banner:js=" <> banner
+        , "--loader:.node=file" -- See https://github.com/evanw/esbuild/issues/1051
+        , format
+        ]
+      , opts.extraArgs
+      , minify
+      , sourceMap
+      , entrypoint
+      ]
+
+  unless opts.forceOverwrite
+    $ whenM (isNotSpagoGeneratedFile outfile)
+    $ die [ "Target file " <> opts.outfile <> " was not previously generated by Spago. Use --force-overwrite to overwrite anyway." ]
```

Comment:
Let's make this change be not breaking in the immediate term by commenting out this check for now, and reintroducing it 6 months down the line. In this way as folks upgrade to the new versions they will start producing bundles with the banner, which will pass the test once we put it back in.

We should leave the tests in too (fixtures and everything), and comment them out too, so they are ready to go when the time comes

```suggestion
  _refuseToOverwriteBundle <- isNotSpagoGeneratedFile outfile
  -- when (refuseToOverwriteBundle && not opts.forceOverwrite) do
  --   die [ "Target file " <> opts.outfile <> " was not previously generated by Spago. Use --force to overwrite anyway." ]
```

Reply:
I'm worried that the tests and/or the main code may rot over time from disuse.

How about instead make this check dependent on the presence of a special marker file, and then make the tests write that file so that the check will run during tests.

In addition to the marker file, we could make the check dependent on time (e.g. start running it in 2025) or on current version (e.g. start running after 1.0.0). We'd still have to have the marker file for the tests though.

Reply:
I added the marker file logic. Let me know if you would like the time or version condition.

Reply:
I wouldn't like to create more files. Let's make this dependent on the spago version (+5 patch versions from now or `1.0.0`, whichever comes first) instead. We won't be able to test this now, but we can already check for the watermark, and we can have tests break once we get past the release threshold - I think it's good enough to prevent rot

Reply:
Oh I see - for some reason I entirely missed that comment. Makes sense now!